packages/headless/src/features/configuration/configuration-state.ts

```diff
@@ -175,7 +175,7 @@ export interface CoreAnalyticsState {
    * The unique identifier of the tracking target.
    * @internal
    */
-  trackingId: string;
+  trackingId?: string;
```

Comment:
@fbeaudoincoveo For the commerce engine, would it be considered a breaking change to make this type optional, and later bring it back to being required? We would still keep the validation that ensures a `trackingId` is provided when building a commerce engine.
https://github.com/coveo/ui-kit/blob/904bdaceda272409c418e2aa15e3d96de3066ca1/packages/headless/src/app/commerce-engine/commerce-engine-configuration.ts#L58-L59

Reply:
For Commerce, the state & by extension its shape is/should be purely internal, so any changes to them are non-breaking